src/client/pages/Embark/index.tsx

```diff
@@ -121,7 +120,7 @@ const Embark: React.FunctionComponent<EmbarkProps> = (props) => {
         embarkPassageName: props.name,
       })
     }
-  }, [currentPassage.id])
+  }, [currentPassage, history, state, props.baseUrl, props.name])
```

Comment:
why do we remove `currentPassage.id` here though? What if the id changes but not the `currentPassage`? Then `useEffect` wont detect that change

Reply:
I changed it since the `useEffect` also uses `currentPassage.url`, `currentPassage.api` and `currentPassage.externalRedirect` and instead of adding all of them to the dependency array I thought I could just added `currentPassage`. That might be a mistake though since I guess React will think `currentPassage` is new on every render since it's an object, and run the effect all the time.

Reply:
With that said, I'm pretty sure I checked that the number of renders did not change with this modification. I mean, I've of course manually tested every component I've made changes to, to see that behaviour didn't change.

Reply:
I'm changing it to have `currentPassage.id`, `currentPassage.url`, `currentPassage.api` and `currentPassage.externalRedirect` in the dependency array instead of just `currentPassage` (though I'm not really able to detect any difference in behaviour either way, but I might of course have missed something).